scripts/reset.js

```diff
@@ -0,0 +1,26 @@
+const ADB = require('appium-adb');
+const B = require('bluebird');
+const { flatMap } = require('lodash');
+
+const SERVER_PKGS = [
+  'io.appium.uiautomator2.server',
+  'io.appium.uiautomator2.server.test',
+  'io.appium.settings',
+];
+
+async function runReset () {
+  const adb = new ADB();
+  const udids = (await adb.getConnectedDevices())
+    .filter(({state}) => state === 'device')
+    .map(({udid}) => udid);
+  if (0 === udids.length) {
+    return;
+  }
+  const uninstallPromises = udids.map((udid) => {
+    adb.curDeviceId = udid;
+    return SERVER_PKGS.map((pkgId) => adb.uninstallApk(pkgId));
+  });
+  await B.all(flatMap(uninstallPromises));
+}
+
+(async () => await runReset())();
```

Comment:
might want to do:

```js
const {asyncify} = require('asyncbox')
// ...
asyncify(runReset)
```

to make sure errors are caught/logged/thrown appropriately

Reply:
or even just

```
runReset().catch((err) => throw err)
```

Reply:
isn't asyncify doing the same?

Reply:
it might be, i forget if it does something special like log to stderr. i usually just do `fn().catch(err => throw err)` nowadays.

Reply:
linter is unhappy about that syntax

Reply:
gotta put braces around `throw` i guess, just do whatever makes lint happy.

Reply:
🤷 
For me it is just working as is.
I don't know what else to change

Reply:
leaving as-is will create unhandled rejections.

try this:

```js
runReset().catch((err) => {  throw err; });
```

Reply:
I've deliberately added `throw` to the function and got the following result with the current implementation:

```
/Users/elf/code/appium-uiautomator2-driver/scripts/reset.js:12
  throw new Error('yolo');
        ^
Error: yolo
    at runReset (/Users/elf/code/appium-uiautomator2-driver/scripts/reset.js:12:9)
    at /Users/elf/code/appium-uiautomator2-driver/scripts/reset.js:31:20
    at Object.<anonymous> (/Users/elf/code/appium-uiautomator2-driver/scripts/reset.js:31:31)
    at Module._compile (node:internal/modules/cjs/loader:1120:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1174:10)
    at Module.load (node:internal/modules/cjs/loader:998:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```

where an unhandled exception is observed?

Reply:
ok, guess i'm wrong about that! think we're good to go then